### DISCUSSION TITLE
Is it possible to define default connections during deployment?

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#40955] [Q&A] [answered]
Is it possible to define default connections during deployment?

[This option](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-database-load-default-connections) is deprecated but I don't understand what the alternative is ?

I'm trying to load my own pre-established connections that my DAGs depend on. On different environments (dev/qa/prod)

**Comment by omkar-foss** [comment]
Hey @pedro-cf, yes you're right, the `load_default_connections` option was used with `airflow db init`, which is now deprecated.

You can use it's alternative, the `airflow connections create-default-connections` command which offers the same functionality and creates all default connections from all providers (docs [here](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#create-default-connections)).

Also for more info on the deprecation, kindly note this Deprecation Warning for `load_default_connection` option (given [here](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-database-load-default-connections)):
>Deprecated since version 2.7.0: This option is only used by the deprecated “airflow db init” command. This option has been used in previous versions of Airflow to determine if loading of the default connections is done with the airflow db init command. This command has been deprecated and replaced by two separate commands airflow db migrate and airflow connections create-default-connections and load_default_connections is not used anymore by those commands.

**Reply by pedro-cf** [reply, marked as answer]
I'm refering to loading real connections on startup.

I ended up creating my own plugin that loads a json file at: `/opt/airflow/config/connections.json` and then deletes it using the cli:

```bash
airflow connections import /opt/airflow/config/connections.json
```

**Reply by omkar-foss** [reply]
Okay yes, since you'd mentioned "default" connections in your question, I pointed you to the `airflow connections create-default-connections` which is for initializing default connections.

Indeed, for importing connections in realtime, `airflow connections import <connections-json-file>` should work.

**But you might want to review and ensure that the connections json file is removed with complete certainty after the import into Airflow is done, as leaving it in there could result in security or compliance problems, especially if it contains credentials for production or other critical environments.**

Also, if you're looking for a more secure and dynamic way to update connection credentials in realtime, I'd suggest that you explore the [AWS Secrets Manager storage backend for Airflow](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/secrets-backends/aws-secrets-manager.html), could be very useful for production and other critical environments.

**Reply by potiuk** [reply]
> I'm refering to loading real connections on startup.
> I ended up creating my own plugin that loads a json file at: /opt/airflow/config/connections.json and then deletes it using the cli:

> airflow connections import /opt/airflow/config/connections.json


It's a good way. The CLI is foreseen for that use case and you are free to import/export connections as you want with your own deployment options.  Loading connections at startup means that you will overwrite them when you start airflow - which means that your "manual" changes are lost. But if you want to make it your behaviour and you are OK with maintainig your connections in a file exclusively and not use DB configuration UI - that's your choice and running such script or plugin or deployment script is fine if you know what you are doing.
